src/core/CCList.ml

```diff
@@ -1353,3 +1353,58 @@ let pp ?(start="") ?(stop="") ?(sep=", ") pp_item fmt l =
         (CCFormat.hbox(CCList.pp ~start:"[" ~stop:"]" CCFormat.int)) \
         [1;2;3])
 *)
+
+let split l = 
+  let rec direct i l = match l with 
+    | [] -> ([],[])
+    | [(x1,y1)] -> ([x1], [y1])
+    | [i1; i2] -> 
+      let (x1, y1) = i1 in 
```

Comment:
No need for so many let bindings:

    | [x1, y1; x2, y2] -> [x1; y1], [x2; y2]

etc.